src/Character/Investigator.java

```diff
@@ -1,75 +1,260 @@
-	 * is that the exact role of his target is revealed.
-	 */
-	public Investigator(Player player) {
-		super(Roles.Investigator, player);
-		detectsRole = GameOptions.InvestigatorOptions.DetectRole.DETECT_VAGUE_ROLE;
-	}
+	private Random random_generator;
 
-	/**
-	 * Investigator. Game Option constructor.
-	 * 
-	 * @param gameOption
-	 *            - Investigator.Options.DETECT_EXACT_ROLE: the investigator
-	 *            will discover the exact role of his target -
-	 *            Investigator.Options.DETECT_VAGUE_ROLE: the investigator will
-	 *            discover a hint for the role of his target
-	 */
-	Investigator(Player player, GameOptions.Options detectsRole) {
+	protected Investigator(Player player) {
 		super(Roles.Investigator, player);
-		this.detectsRole = detectsRole;
+		random_generator = new Random();
 	}
 
 	@Override
-	public String doAction() throws CannotGetPlayerException {
-		
-		if (getTarget().size() != 1) {
-			return GameMessage.NO_ACTION;
+	public boolean performAction(List<Player> alive_player, Character target) {
+		if (this.character_status.isBlocked()) {
+			result = GameLog.Character.BLOCKED(player);
+			return true;
 		}
-		
-		Player target = getTarget().get(0);
-		
-		if(!GameEngine.getCharacter(target).isAlive()) {
-			return GameMessage.TARGET_DEAD;
+		target.character_status.getVisitors().add(player);
+		List<Investigations> investigation_list = Investigations
+				.doInvestigation(target.getRole());
+		int index = random_generator.nextInt(investigation_list.size());
+		result = GameLog.Character.INVESTIGATION_SUCCESS(player,
+				target.player, investigation_list.get(index).toString());
+		return true;
+	}
+
+	private enum Investigations {
+		Trespassing("Trespassing"), Kidnapping("Kidnapping"), NoCrime(
+				"No Crime"), Corruption("Corruption"), IdentityThief(
+				"Identity Thief"), Soliciting("Soliciting"), Murder("Murder"), DisturbanceOfPeace(
+				"Disturbance of Peace"), Conspiracy("Conspiracy"), DestructionOfProperty(
+				"Destruction of Property"), Arson("Arson");
+
+		String investigation;
+
+		Investigations(String investigation) {
+			this.investigation = investigation;
 		}
-		if (this.isRoleBlocked()) {
-			return GameMessage.ROLE_BLOCKED;
+
+		@Override
+		public String toString() {
+			return investigation.toLowerCase();
 		}
-		
-		GameEngine.getCharacter(target).addVisitor(getPlayer());
-		
-		DetectRole detect = (DetectRole) detectsRole;
-		
-		switch (detect) {
-		case DETECT_EXACT_ROLE:
-			return GameMessage.INVESTIGATION_EXACT(target.getName());
-		case DETECT_VAGUE_ROLE:
-			return GameMessage.INVESTIGATION_VAGUE(target.getName());
-		default:
-			return GameMessage.INVESTIGATION_VAGUE(target.getName());
+
+		/**
+		 * Makes a list of possible investigations for a role
+		 * 
+		 * @param role
+		 * @return list of possible investigation results for that role
+		 */
+		protected static List<Investigations> doInvestigation(Roles role) {
```

Comment:
Yeah so, Spring framework/dependency injection in the most simple terms is:
Say you have a class car that has a class wheel that has a class rubber.
The traditional way to do it in java is to do new rubber and wrap it in new wheel, and wrap it in new car.
The spring framework lets you instead give it an id of any particular configuration of car, and with only one line, you get new car with a wheel with rubber.
The blueprint for making that new car is in a .xml
So for the case of investigator and consigliere, they can use the same .xml (same source).
I don't think you want to put sheriff with investigator though. Each should hide their own abstraction.
Just a suggestion though so yeah, if there's any implementation you want to try, just do it and make a pull request!